### PR TITLE
BUG: fix valueError for groupby with empty inputs when sample method is called

### PR DESCRIPTION
**Pull request by ntachukwu**
- [x] closes ##48459
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).

### CODE REVIEW

**Comment by mroeschke** [comment]
Looks pretty good. Could use a whatsnew note in `1.6.0.rst` under the groupby section.

**Review by rhshadrach — CHANGES_REQUESTED** [review]
Looks good, small request

File: pandas/tests/groupby/test_sample.py [hunk 1]
```diff
@@ -142,3 +142,10 @@ def test_groupby_sample_with_selections():
     result = df.groupby("a")[["b", "c"]].sample(n=None, frac=None)
     expected = DataFrame({"b": [1, 2], "c": [1, 2]}, index=result.index)
     tm.assert_frame_equal(result, expected)
+
+
+def test_groupby_sample_with_empty_inputs():
+    result = DataFrame({"a": [], "b": []})
+    expected = result.groupby("a").sample()
```
Comment (rhshadrach) [inline pandas/tests/groupby/test_sample.py:149]: nit: `result` should be the result that is being tested, expected should be what we expect. Can you switch these?

**Review by rhshadrach — CHANGES_REQUESTED** [review]
Forgot to submit the request.

File: pandas/tests/groupby/test_sample.py [hunk 2]
```diff
@@ -142,3 +142,15 @@ def test_groupby_sample_with_selections():
...
+def test_groupby_sample_with_empty_inputs():
+    # GH48459
+    df = DataFrame({"a": [], "b": []})
+    groupby_df = df.groupby("a")
+
+    result = groupby_df.sample()
+    expected = df
+    tm.assert_frame_equal(result, expected)
+
+    tm.assert_frame_equal(result, expected)
```
Comment (rhshadrach) [inline pandas/tests/groupby/test_sample.py:156]: Change looks good - you just wound up with a duplicated line.

**Review by rhshadrach — CHANGES_REQUESTED** [review]

File: pandas/tests/groupby/test_sample.py (hunk 2 above)
Comment (Th3nn3ss) [reply pandas/tests/groupby/test_sample.py:156]: Sorry about that. Not sure now that happend.

**Review by rhshadrach — APPROVED** [review]
lgtm

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @Th3nn3ss

**Pull request merged**